### ISSUE TITLE
RuntimeException: PHP Error: Undefined property: stdClass::$range

### ISSUE THREAD

**Issue by jgmdev** [vimeo/psalm#10353]
**PHP Version**: 8.2.12

Was testing the language server component of this project with [Pragtical](https://github.com/pragtical/pragtical) text editor + [LSP](https://github.com/pragtical/lsp) plugin and sending the document changed event triggered the following error:

```
[psalm-language-server] RuntimeException: PHP Error: Undefined property: stdClass::$range in vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/Server/TextDocument.php:122 for command with CLI args "vendor/bin/psalm-language-server --config psalm.xml" in vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:77
```
Fixed it locally with the following change, but not sure if it is the proper solution:

vendor/vimeo/psalm/src/Psalm/Internal/LanguageServer/Server/TextDocument.php
```diff
-        if (count($contentChanges) === 1 && isset($contentChanges[0]) && $contentChanges[0]->range === null) {
+        if (count($contentChanges) === 1 && isset($contentChanges[0]) && (!property_exists($contentChanges[0], "range") || $contentChanges[0]->range === null)) {
```

**Comment by jgmdev** [comment]
> Hey @jgmdev, can you reproduce the issue on https://psalm.dev ?

yes

**Comment by jgmdev** [comment]
> Is there any chance you're running PHP with [`opcache.save_comments=0`](https://www.php.net/manual/en/opcache.configuration.php#ini.opcache.save-comments)?

You are correct that is the case on my php opcache config. If not supported by psalm should this issue get closed?

**Comment by weirdan** [comment]
No, we should detect this configuration and warn about / correct it.

**Comment by weirdan** [comment]
Fixed in `5.x-dev` (`composer require --dev vimeo/psalm:5.x-dev`)
